src/IO/ReadHelpers.h

```diff
@@ -706,7 +706,7 @@ inline void convertToDayNum(DayNum & date, ExtendedDayNum & from)
 }
 
 template <typename ReturnType = void>
-inline ReturnType readDateTextImpl(DayNum & date, ReadBuffer & buf)
+inline ReturnType readDateTextImpl(DayNum & date, ReadBuffer & buf, const DateLUTImpl & date_lut = DateLUT::instance())
```

Comment:
Functions `readDateTextImpl()` are helper methods and not supposed to be called from outside file ReadHelpers.h. We should make argument `date_lut` mandatory (the little bit of extra convenience is not needed).